python/ray/_private/metrics_agent.py

```diff
@@ -187,12 +275,61 @@ def _record_metrics(self, metrics):
                         )
                     else:
                         raise ValueError("Summary is not supported")
-
                     view_data.tag_value_aggregation_data_map[tag_vals] = data
 
-        # Finally, export all the values
+        # Finally, export all the values.
+        # When the view data is exported, they are hard-copied.
         self.view_manager.measure_to_view_map.export(view_data_changed)
 
+    def clean_all_dead_worker_metrics(self):
+        """Clean dead worker's metrics.
+
+        Worker metrics are cleaned up if there was no metrics
+        report for more than `worker_timeout_s`.
+
+        This method has to be periodically called by a caller.
+        """
+        with self._lock:
+            if not self.view_manager:
+                return
+
+            worker_ids_to_clean = []
+            for worker_id_hex, state in self.worker_id_to_state.items():
+                if time.time() - state.last_reported_time > self.worker_timeout_s:
+                    worker_ids_to_clean.append(worker_id_hex)
+
+            for worker_id in worker_ids_to_clean:
+                self._clean_worker_metrics(worker_id)
```

Comment:
Add some log message that we're cleaning data for this worker?

Reply:
good idea! Added